tedana/viz.py

```diff
@@ -0,0 +1,257 @@
+"""
+Functions to creating figures to inspect tedana output
+"""
+import logging
+import os
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from tedana import model
+
+LGR = logging.getLogger(__name__)
+
+
+def writecompfigs(ts, mask, comptable, mmix, n_vols,
+                  acc, rej, midk, empty, ref_img):
+    """
+    Creates static figures that highlight certain aspects of tedana processing
+    This includes figure for each component showing the component time course,
+    the spatial weight map and a fast Fourier transform of the time course
+
+    Parameters
+    ----------
+    ts : (S x T) array_like
+        Time series from which to derive ICA betas
+    mask : (S,) array_like
+        Boolean mask array
+    comptable : (N x 5) array_like
+        Array with columns denoting (1) index of component, (2) Kappa score of
+        component, (3) Rho score of component, (4) variance explained by
+        component, and (5) normalized variance explained by component
+    mmix : (C x T) array_like
+        Mixing matrix for converting input data to component space, where `C`
+        is components and `T` is the same as in `data`
+    n_vols : :obj:`int`
+        Number of volumes in original time series
+    acc : :obj:`list`
+        Indices of accepted (BOLD) components in `mmix`
+    rej : :obj:`list`
+        Indices of rejected (non-BOLD) components in `mmix`
+    midk : :obj:`list`
+        Indices of mid-K (questionable) components in `mmix`
+    empty : :obj:`list`
+        Indices of ignored components in `mmix`
+    ref_img : :obj:`str` or img_like
+        Reference image to dictate how outputs are saved to disk
+
+    """
+
+    # regenerate the beta images
+    ts_B = model.get_coeffs(ts, mmix, mask)
+    ts_B = ts_B.reshape(ref_img.shape[:3] + ts_B.shape[1:])
+    # Mask out zeros
+    ts_B = np.ma.masked_where(ts_B == 0, ts_B)
+
+    # Get repitition time from ref_img
+    tr = ref_img.header.get_zooms()[-1]
+
+    # Start making plots
+    if not os.path.exists('figures'):
+        os.mkdir('figures')
+
+    # Create indices for 6 cuts, based on dimensions
+    cuts = [ts_B.shape[dim] // 6 for dim in range(3)]
+
+    for compnum in range(0, mmix.shape[1], 1):
+
+        allplot = plt.figure(figsize=(10, 9))
```

Comment:
For really long series, with many components to plot, this is going to end up consuming a lot of memory. Instead of instantiating a new figure in each loop, can we just update it ? See [this StackOverflow post](https://stackoverflow.com/a/38980600) for a little more detail.

Reply:
This one is going to take a bit longer to resolve, given my lack of understanding about matplotlib, and the mix of subplots going on...